src/platform/Darwin/DnssdContexts.cpp

```diff
@@ -354,6 +354,26 @@ CHIP_ERROR ResolveContext::OnNewAddress(uint32_t interfaceId, const struct socka
     ChipLogDetail(Discovery, "Mdns: %s interface: %" PRIu32 " ip:%s", __func__, interfaceId, addrStr);
 #endif // CHIP_DETAIL_LOGGING
 
+    auto flags         = 0;
+    auto name          = interfaces[interfaceId].hostNameWithDomain.c_str();
+    auto rrtype        = ip.IsIPv4() ? kDNSServiceType_A : kDNSServiceType_AAAA;
+    auto rrclass       = kDNSServiceClass_IN;
+    auto rdlen         = static_cast<uint16_t>(ip.IsIPv4() ? sizeof(in_addr) : sizeof(in6_addr));
+    const void * rdata = nullptr;
+    if (ip.IsIPv4())
+    {
+        rdata = &(((struct sockaddr_in *) address)->sin_addr);
+    }
+    else
+    {
+        rdata = &(((struct sockaddr_in6 *) address)->sin6_addr);
+    }
+
+    // It may happens that there is stale DNS entry. If so, DNSServiceReconfirmRecord will flush the record from the daemon cache
+    // once it will determine that it is invalid. It may not help for this particular resolve, but may help subsequent resolves.
```

Comment:
```suggestion
    // It may happen that there is stale DNS entry. If so, DNSServiceReconfirmRecord will flush the record from the daemon cache
    // once it determines that it is invalid. It may not help for this particular resolve, but may help subsequent resolves.
```